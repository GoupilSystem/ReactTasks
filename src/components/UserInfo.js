import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ isAuthenticated, user, login, logout }) => {
  if (isAuthenticated && user) {
    return (
      <div>
        <p>Welcome, {user.name}</p>
        <p>Email: {user.email}</p>
        <button onClick={logout}>Logout</button>
      </div>
    );
  } else {
    return (
      <p>
        <button onClick={login}>Login</button>
      </p>
    );
  }
};

UserInfo.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }),
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
};

export default UserInfo;
