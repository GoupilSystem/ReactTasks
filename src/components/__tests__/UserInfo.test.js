import React from 'react';
import { render, screen } from '@testing-library/react';
import UserInfo from '../UserInfo';

// test('renders login button when not authenticated', () => {
//   // Create mock login and logout functions
//   const login = jest.fn();
//   const logout = jest.fn();

//   // Render the component in the non-authenticated state
//   render(<UserInfo isAuthenticated={false} user={{}} login={login} logout={logout} />);

//   // Assert that the login button is displayed
//   expect(screen.getByText('Login')).toBeInTheDocument();
// });

// test('renders user info when authenticated', () => {
//     // Create mock login and logout functions
//     const login = jest.fn();
//     const logout = jest.fn();
  
//     // Mock user data
//     const user = {
//       name: 'John Doe',
//       username: 'john.doe@example.com'
//     };
  
//     // Render the component in the authenticated state
//     render(<UserInfo isAuthenticated={true} user={user} login={login} logout={logout} />);
  
//     // Assert that the user name is displayed
//     expect(screen.getByText(/Welcome, John Doe/i)).toBeInTheDocument();
//   });

  // Snapshot test
test('renders UserInfo component correctly', () => {
    // Create mock user data
    const user = {
      name: 'John Doe',
      username: 'john.doe@example.com',
    };
  
    // Render the component
    const { asFragment } = render(<UserInfo isAuthenticated={true} user={user} login={() => {}} logout={() => {}} />);
    
    // Assert that the component's UI matches the stored snapshot
    expect(asFragment()).toMatchSnapshot();
  });