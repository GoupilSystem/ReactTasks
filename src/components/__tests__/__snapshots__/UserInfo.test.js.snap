// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserInfo component correctly 1`] = `
<DocumentFragment>
  <div
    style="display: flex; align-items: center;"
  >
    <p
      style="margin-right: 10px;"
    >
      Welcome, John Doe
    </p>
    <p
      style="margin-right: 10px;"
    >
      Email: john.doe@example.com
    </p>
    <button>
      Logout
    </button>
  </div>
</DocumentFragment>
`;
